homeassistant/components/homewizard/sensor.py

```diff
@@ -0,0 +1,208 @@
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="active_power_w",
+        name="Active Power",
+        native_unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="active_power_l1_w",
+        name="Active Power L1",
+        native_unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="active_power_l2_w",
+        name="Active Power L2",
+        native_unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="active_power_l3_w",
+        name="Active Power L3",
+        native_unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="total_gas_m3",
+        name="Total Gas",
+        native_unit_of_measurement=VOLUME_CUBIC_METERS,
+        device_class=DEVICE_CLASS_GAS,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="gas_timestamp",
+        name="Gas Timestamp",
+        device_class=DEVICE_CLASS_TIMESTAMP,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Initialize sensors."""
+    coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        COORDINATOR
+    ]
+
+    entities = []
+    if coordinator.api.data is not None:
+        for description in SENSORS:
+            if description.key in coordinator.api.data.available_datapoints:
+                entities.append(HWEnergySensor(coordinator, entry, description))
+    async_add_entities(entities)
+
+
+class HWEnergySensor(CoordinatorEntity[DeviceResponseEntry], SensorEntity):
+    """Representation of a HomeWizard Sensor."""
+
+    def __init__(
+        self,
+        coordinator: HWEnergyDeviceUpdateCoordinator,
+        entry: ConfigEntry,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize Sensor Domain."""
+
+        super().__init__(coordinator)
+        self.entity_description = description
+        self.entry = entry
+
+        # Config attributes.
+        self._attr_name = f"{entry.title} {description.name}"
+        self.data_type = description.key
+        self._attr_unique_id = f"{entry.unique_id}_{description.key}"
+
+        # Some values are given, but set to NULL (eg. gas_timestamp when no gas meter is connected)
+        if self.data["data"][self.data_type] is None:
+            self.entity_description.entity_registry_enabled_default = False
```

Comment:
the entity descriptions are part of constants. If a user would set up 2 Homewizard devices, it is now overriden. Make a copy instead.

Reply:
Although a better idea would be to not add these entities to begin with ? They will never have data.

Reply:
Changed it that `None` values won't be added (c593306). The `power_export` enabled_default are now set into `_attr_entity_registry_enabled_default` (a1be964)